examples/media/src/widgets/Dates.ts

```diff
@@ -0,0 +1,50 @@
+import { numericMenu, panel } from 'instantsearch.js/es/widgets';
+import {
+  startOfDay,
+  endOfDay,
+  startOfWeek,
+  endOfWeek,
+  startOfMonth,
+  endOfMonth,
+  startOfYear,
+  endOfYear,
+  getUnixTime,
+} from 'date-fns';
+
+const currentDate = new Date();
+
+const createDatesList = header =>
+  panel({
+    templates: {
+      header,
+    },
+  })(numericMenu);
+
+export const createDates = ({ container, header }) =>
+  createDatesList(header)({
+    container,
+    attribute: 'created_at_timestamp',
+    items: [
+      { label: 'All time' },
+      {
+        label: 'Today',
+        start: getUnixTime(startOfDay(currentDate)),
+        end: getUnixTime(endOfDay(currentDate)),
+      },
+      {
+        label: 'This week',
+        start: getUnixTime(startOfWeek(currentDate)),
+        end: getUnixTime(endOfWeek(currentDate)),
+      },
+      {
+        label: 'This month',
+        start: getUnixTime(startOfMonth(currentDate)),
```

Comment:
I first thought this didn't work, but as it's the beginning of the month, there were only a few days for items to show up. Maybe the items should be "today", "last 7 days", "last 30 days", "last year"?

Reply:
Haha this puzzled me as well for a bit, as I also expected a "last X" behavior. I could borrow the wording from Hacker News, as it's used widely in production. Wdyt?

![image](https://user-images.githubusercontent.com/154633/140372552-7bc3fd70-3ca0-4539-b6dc-bdb5d23d48cb.png)

Reply:
yes, that wording & behaviour seems more consistent